cwatch.php

```diff
@@ -827,63 +829,70 @@ private function getSitesTab($template, $service, $post)
-                $scanner_errors = $scanner->errors();
-                if (empty($scanner_errors)) {
-                    $site->scanner = $scanner->response();
-                }
-
-                // Get the license attached to this domain
-                $license = $api->getLicense($site->licenseKey);
-                if (empty($scanner_errors)) {
-                    $site->license = $license->response();
-                }
+        $sites_response = $api->getSites($service_fields->cwatch_email);
+        $sites_errors = $sites_response->errors();
+        if (empty($sites_errors)) {
+            $sites = $sites_response->response();
+        }
 
-                $taken_licenses[] = $site->licenseKey;
-                $sites[] = $site;
+        // Sort sites by license
+        $sites_by_license = [];
+        foreach ($sites as $site) {
+            // Get the malware scanner for this site
+            $scanner = $api->getScanner($service_fields->cwatch_email, $site->domain);
+            $scanner_errors = $scanner->errors();
+            if (empty($scanner_errors)) {
+                $site->scanner = $scanner->response();
             }
+
+            $sites_by_license[$site->licenseKey] = $site;
         }
 
         // Get cWatch licenses
         $licenses_response = $api->getLicenses($service_fields->cwatch_email);
         $licenses_errors = $licenses_response->errors();
         $licenses = [];
+        $available_licenses = [];
         if (empty($licenses_errors)) {
             foreach ($licenses_response->response() as $license) {
-                if (strtolower($license->status) == 'valid'
-                    && $license->registeredDomainCount == 0
-                    && !in_array($license->licenseKey, $taken_licenses)
-                ) {
-                    $licenses[$license->licenseKey] = $license->productTitle;
+                if (isset($sites_by_license[$license->licenseKey])) {
+                    // Use the associated site for domain info
+                    $license->site = $sites_by_license[$license->licenseKey];
+                } elseif (isset($provisions_by_license[$license->licenseKey])) {
+                    // Use the associated provision for site info
+                    $license->site = $provisions_by_license[$license->licenseKey];
+                } else {
+                    // Mark this license available for attaching a new domain
+                    $available_licenses[$license->licenseKey] = $license->productTitle;
                 }
+
+                $licenses[] = $license;
             }
         }
 
         $this->view->set('site_statuses', $this->getSiteStatuses());
-        $this->view->set('sites', $sites);
         $this->view->set('licenses', $licenses);
+        $this->view->set('available_licenses', $available_licenses);
         $this->view->set('service', $service);
+        $this->view->set('license_key', isset($get['key']) ? $get['key'] : '');
```

Comment:
It doesn't look to me like this license_key is being used on the tab templates?

Reply:
It is in the add_site templates